### PR TITLE
NOISSUE: setup cache for go mod based on go.sum

### PR DESCRIPTION
**Pull request by sexyhamster**
What I did:
- removed testsum
- added cache for go modules folder
- for docker images vendor is created outside of image, so cache will be used

### CODE REVIEW

File: ledger-core/test.log [hunk 1]
```diff
@@ -0,0 +1,11096 @@
...benchs github.com/insolar/assured-ledger/ledger-core/virtual/integration/convlog github.com/insolar/assured-ledger/ledger-core/virtual/integration/deduplication github.com/insolar/assured-ledger/ledger-core/virtual/integration/mock/publisher github.com/insolar/assured-ledger/ledger-core/virtual/integration/mock/publisher/checker github.com/insolar/assured-ledger/ledger-core/virtual/integration/mock/publisher/cmd/insolar-mockgenerator-typedpublisher github.com/insolar/assured-ledger/ledger-core/virtual/integration/utils github.com/insolar/assured-ledger/ledger-core/virtual/object github.com/insolar/assured-ledger/ledger-core/virtual/object/finalizedstate github.com/insolar/assured-ledger/ledger-core/virtual/pulsemanager github.com/insolar/assured-ledger/ledger-core/virtual/statemachine github.com/insolar/assured-ledger/ledger-core/virtual/testutils github.com/insolar/assured-ledger/ledger-core/virtual/testutils/shareddata github.com/insolar/assured-ledger/ledger-core/virtual/testutils/slotdebugger -args TESTLOG_OUT="/Users/ilyaozherelyev/go/src/github.com/insolar/assured-ledger/ledger-core/integration_extended_test.log" TESTLOG_MARKS=1 TESTLOG_TXT=1 2>&1 | tee "integration_test.log"
```
Comment (AndreyBronin) [inline ledger-core/test.log:1]: .gitignore or remove this

Comment (sexyhamster) [reply ledger-core/test.log]: removed

File: .github/workflows/linux-checks.yaml [hunk 2]
```diff
@@ -30,10 +29,17 @@ jobs:
...
+          path: ${{env.GOPATH}}/pkg/mod
+          key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
+          restore-keys: |
+            ${{ runner.os }}-go-
       - name: Fetch vendor and check succesful build
         run: |
-          go mod download
-          make vendor install-build-tools pre-build proto build
+          make vendor
+          make install-build-tools pre-build proto build
```
Comment (ita-sammann) [inline .github/workflows/linux-checks.yaml:42]: Why move vendor target to a separate line?

File: .github/workflows/linux-checks.yaml [hunk 3]
```diff
@@ -64,10 +70,19 @@ jobs:
...
+          path: ${{env.GOPATH}}/pkg/mod
+          key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
+          restore-keys: |
+            ${{ runner.os }}-go-
       - name: Fetch vendor and check succesful build
         run: |
           go mod download
-          make vendor install-build-tools build
+          make vendor
+          go install -v github.com/cyraxred/go-acc
```
Comment (ita-sammann) [inline .github/workflows/linux-checks.yaml:84]: WTF? Why it is not in our dependencies?

Comment (sexyhamster) [reply .github/workflows/linux-checks.yaml:84]: It is, but we don't need here all the build tools, just go-acc

File: .github/workflows/linux-checks.yaml (hunk 2 above)
Comment (sexyhamster) [reply .github/workflows/linux-checks.yaml]: fixed

File: .github/workflows/linux-checks.yaml (hunk 3 above)
Comment (sexyhamster) [reply .github/workflows/linux-checks.yaml]: switched to common install-build-tools

File: .github/workflows/linux-checks.yaml [hunk 4]
```diff
@@ -154,18 +172,34 @@ jobs:
...
+            ${{ runner.os }}-go-
+      - name: Fetch dependencies
+        run: |
+          go mod download
+          make vendor
+        working-directory: ${{env.V2_PATH}}
       - name: build images
-        run: make docker-build
+        run: |
+          make docker-build
```
Comment (ita-sammann) [inline .github/workflows/linux-checks.yaml:189]: ?

File: .github/workflows/linux-checks.yaml [hunk 5]
```diff
@@ -154,18 +172,34 @@ jobs:
...
         working-directory: ${{env.V2_PATH}}
       - name: push assured ledger image to local registry
         run: |
           docker tag insolar/assured-ledger:latest localhost/assured-ledger:latest
           docker push localhost/assured-ledger:latest
+      - name: build tools
+        run: |
+          make install-build-tools
```
Comment (ita-sammann) [inline .github/workflows/linux-checks.yaml:197]: ?

File: .github/workflows/linux-checks.yaml [hunk 6]
```diff
@@ -64,10 +72,20 @@ jobs:
...
+          restore-keys: |
+            ${{ runner.os }}-go-
+      - name: Fetch dependencies
         run: |
           go mod download
-          make vendor install-build-tools build
+          make vendor
+      - name: Prepare go-acc
+        run: |
+          make install-build-tools
```
Comment (ita-sammann) [inline .github/workflows/linux-checks.yaml:88]: ?

Comment (sexyhamster) [reply .github/workflows/linux-checks.yaml]: fixed

File: .github/workflows/linux-checks.yaml (hunk 5 above)
Comment (sexyhamster) [reply .github/workflows/linux-checks.yaml]: fixed

File: .github/workflows/linux-checks.yaml (hunk 4 above)
Comment (sexyhamster) [reply .github/workflows/linux-checks.yaml]: fixed

**Pull request merged**